lib/kino_db/connection_cell.ex

```diff
@@ -141,6 +162,12 @@ defmodule KinoDB.ConnectionCell do
     not Enum.any?(keys, fn key -> attrs[key] in [nil, ""] end)
   end
 
+  defp conditional_fields_filled?(_, []), do: true
+
+  defp conditional_fields_filled?(attrs, keys) do
+    not Enum.all?(keys, fn key -> attrs[key] in [nil, ""] end)
```

Comment:
In this case it's a micro-optimisation, but generally we can halt sooner by reversing the logic :)

```suggestion
    Enum.any?(keys, fn key -> attrs[key] not in [nil, ""] end)
```